bika/lims/exportimport/instruments/resultsimport.py

```diff
@@ -753,6 +753,44 @@ def _getZODBAnalysesFromReferenceAnalyses(self, objid, criteria):
 
         return analyses
 
+    def analysisHasCalcution(self, objid, analysis):
+        """ returns True or False"""
+
+        analyses = self._getZODBAnalyses(objid)
+        analyses_with_calculation = filter(lambda c: c.getCalculation(),analyses)
+        if analyses_with_calculation:
+            return True
+        return False
+
+    def getAnalysesWithCalculation(self, objid, analysis, analyses):
+        if self.analysisHasCalcution(objid, analysis):
+            analyses_with_calculation = filter(lambda c: c.getCalculation(),analyses)
+            return analyses_with_calculation
```

Comment:
This method does the same check code as `analysisHasCalcution` with the difference that it returns the list of analyses instead of casting the list to a boolean. IMO it is not needed at all, because you can do it in `calcluateTotalResults` just at the beginning.